src/components/Pagination/Pagination.tsx

```diff
@@ -1 +1,95 @@
-export const Pagination = () => {};
+import React from 'react';
+import { getNumbers } from '../../utils';
+import classNames from 'classnames';
+
+interface Props {
+  total: number;
+  perPage: number;
+  currentPage: number;
+  onPageChange: (pageNum: number) => void;
+}
+
+export const Pagination: React.FC<Props> = ({
+  total,
+  perPage,
+  currentPage,
+  onPageChange,
+}) => {
+  const maxNumPages = Math.ceil(total / perPage);
+  // eslint-disable-next-line @typescript-eslint/no-unused-vars
+  const items = getNumbers(
+    (currentPage - 1) * perPage + 1,
+    currentPage * perPage < total ? currentPage * perPage : total,
+  ).map((n: number) => `Item ${n}`);
+  const links = getNumbers(1, maxNumPages).map((n: number) => n);
+  const visibleItems = items.slice(0, perPage);
+
+  return (
+    <>
+      <ul className="pagination">
+        <li
+          className={classNames('page-item', { disabled: currentPage === 1 })}
+        >
+          <a
+            data-cy="prevLink"
+            className="page-link"
+            href="#prev"
+            aria-disabled={`${currentPage === 1 ? 'true' : 'false'}`}
```

Comment:
Using `href="#prev"` can cause unwanted behavior in the application by updating the URL hash when clicked. It is recommended to use `event.preventDefault()` in the `onClick` handler to prevent the default anchor behavior.